From 6aa7f63a31ec6bb712245823306b1a976e310018 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 4 Mar 2015 15:26:59 +0800
Subject: [PATCH 0213/5242] MLK-10367 tty: serial: imx: set uart module clock
 to 80Mhz

commit  987295fc169b095dd05c0c0b9ae1dd9d01fecbd6 from
https://source.codeaurora.org/external/imx/linux-imx.git

When UART module clock is great than 80Mhz, there may have risk after
confirming with IC owner. So set the maximum module clock to 80Mhz.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: 330a1245cb91583d9bc916bbb6d8c7d2c86b26f3)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 554a69d..ffa4482 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -176,6 +176,7 @@
 #define DRIVER_NAME "IMX-uart"
 
 #define UART_NR 8
+#define IMX_MODULE_MAX_CLK_RATE	80000000
 
 /* i.MX21 type uart runs on all i.mx except i.MX1 and i.MX6q */
 enum imx_uart_type {
@@ -2256,6 +2257,9 @@ static int imx_uart_probe(struct platform_device *pdev)
 	}
 
 	sport->port.uartclk = clk_get_rate(sport->clk_per);
+	if (sport->port.uartclk > IMX_MODULE_MAX_CLK_RATE)
+		clk_set_rate(sport->clk_per, IMX_MODULE_MAX_CLK_RATE);
+	sport->port.uartclk = clk_get_rate(sport->clk_per);
 
 	/* For register access, we only need to enable the ipg clock. */
 	ret = clk_prepare_enable(sport->clk_ipg);
-- 
1.7.9.5

